src/extension/handler/uri.ts

```diff
@@ -35,24 +36,31 @@ import {
 const REGEX_WEB_RESOURCE = /^https?:\/\//
 const log = getLogger('RunmeUriHandler')
 
-export class RunmeUriHandler implements UriHandler {
+export class RunmeUriHandler extends EventEmitter<Uri> implements UriHandler {
   constructor(
     private context: ExtensionContext,
     private kernel: Kernel,
     private forceNewWindow: boolean,
-  ) {}
+  ) {
+    super()
+  }
 
   async handleUri(uri: Uri) {
     log.info(`triggered RunmeUriHandler with ${uri}`)
     const params = new URLSearchParams(uri.query)
+    const windowId = params.get('windowId')
+    const state = params.get('state')
+    const code = params.get('code')
     const command = params.get('command')
 
     if (!command) {
       window.showErrorMessage('No query parameter "command" provided')
       return
     }
-
-    if (command === 'setup') {
+    if (command === 'auth' && windowId && state && code) {
+      this.fire(uri)
+      return
+    } else if (command === 'setup') {
```

Comment:
@pastuxso, I trust you tested the existing commands to work as expected. Both `setup` and `demo`.

Reply:
@sourishkrout No problem, I tested both commands too :+1: